include/deal.II/base/mpi.templates.h

```diff
@@ -457,6 +457,83 @@ namespace Utilities
 
 
 
+    template <typename T1, typename T2>
```

Comment:
this means I have to include mpi.templates.h to be able to use this class, right? Is that useful? I would move the implementation into a normal header. (and not mpi.h)